### PR TITLE
Documentation for Throttled Replication

### PR DESCRIPTION
**Pull request by benstopford**

### CODE REVIEW

File: docs/ops.html [hunk 1]
```diff
@@ -346,6 +346,127 @@
...
+
+<pre>leader.replication.throttled.replicas
+follower.replication.throttled.replicas</pre>
+
+<p>Which are configured per topic. All four config values are automatically assigned by kafka-reassign-partitions.sh
+    (discussed below). </p>
+<p>To view the throttle limit configuration:</p>
+
+<pre>$ bin/kafka-configs.sh --describe --zookeeper localhost:2181 --entity-type brokers
+Configs for brokers '2' are leader.replication.throttled.rate=1000000,follower.replication.throttled.rate=1000000
```
Comment (junrao) [inline docs/ops.html]: Perhaps use the throttling value consistent with the one used above when executing the tool? Ditto below.

File: docs/ops.html [hunk 2]
```diff
@@ -346,6 +346,127 @@
...
+
+<pre>$ bin/kafka-configs.sh --describe --zookeeper localhost:2181 --entity-type topics
+Configs for topic 'my-topic' are leader.replication.throttled.replicas=1:102,0:101,
+    follower.replication.throttled.replicas=1:101,0:102</pre>
+
+<p>Here we see the leader throttle is applied to partition 1 on broker 102 and partition 0 on broker 101. Likewise the
+    follower throttle is applied to partition 1 on
+    broker 101 and partition 0 on broker 102. </p>
+
+<p>By default kaka-reassign-partitions.sh will apply the leader throttle to all replicas that exist before the
```
Comment (junrao) [inline docs/ops.html]: typo kaka

File: docs/ops.html [hunk 3]
```diff
@@ -346,6 +346,127 @@
...
+<pre>kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)</pre>
+
+<p>The lag should constantly decrease during replication.  If the metric does not decrease the administrator should
+    increase the
+    throttle throughput as described above. </p>
+
+<p><i>(3) Avoiding long delays during replication:</i></p>
+<p>The throttled throughput should be large enough that replicas cannot be starved for extended periods. A good,
+    conservative rule of thumb is
+    to keep throttle above #brokers MB/s where #brokers is the number of brokers in your cluster.  </p>
```
Comment (junrao) [inline docs/ops.html]: Hmm, if there are 100 brokers in the cluster, does that mean the min throttle value has to be 100MB/sec?

File: docs/ops.html [hunk 4]
```diff
@@ -346,6 +346,127 @@
...
+<p>Administrators wishing to use lower throttle values can tune the response size used for replication based on
+    the relation:</p>
+<pre></pre>Worst-Case-Delay = replica.fetch.response.max.bytes x #brokers / throttle</pre>
+
+<p>Here, the admin should tune the throttle  and/or replica.fetch.response.max.bytes  appropriately to ensure the
+    delay is never larger than replica.lag.time.max.ms (as it is possible for some partitions, particularly smaller
+    ones, to enter the ISR before the rebalance completes) or the outer throttle window:
```
Comment (junrao) [inline docs/ops.html]: outer throttle window? Perhaps full throttle window?

File: docs/ops.html (hunk 3 above)
Comment (benstopford) [reply docs/ops.html]: As you're putting that change in i'm inclined to remove the whole section.

**Comment by gwenshap** [comment]
LGTM.
This is a PR on trunk - do we need to cherry-pick to 0.10.1.0? @hachikuji

**Comment by hachikuji** [comment]
@gwenshap May as well. There's another minor doc issue that I'm merging there, so I was planning to regenerate site docs anyway.

**Comment by gwenshap** [comment]
Dual-committed.

**Pull request closed without merging**